File: code/community/1301688526168920065/improving-accuracy-spelled-words-transcriptions.js
// Node.js example for using Deepgram's real-time streaming API with keyword boosting
// Requires the `deepgram-sdk` npm package

const Deepgram = require('@deepgram/sdk');

// Initialize Deepgram with your API key
const deepgramApiKey = process.env.DEEPGRAM_API_KEY;
const deepgram = new Deepgram(deepgramApiKey);

// Options for real-time transcription
const options = {
    punctuate: true,
    keywords: ["JAY", "HAYWIRE"]
};

// Function to handle streaming
async function streamAudio() {
    const response = await deepgram.transcription.live(options);

    response.on('open', () => {
        console.log('Connection opened');
    });

    response.on('transcriptReceived', (data) => {
        console.log('Transcript:', data.channel.alternatives[0].transcript);
    });

    response.on('close', () => {
        console.log('Connection closed');
    });
}

streamAudio();

// Note: Ensure you have a method to send audio data to this script

File: code/community/1301688526168920065/improving-accuracy-spelled-words-transcriptions.py
# Python example for using Deepgram's real-time streaming API with keyword boosting
# Requires the `deepgram-sdk` library

import os
from deepgram import Deepgram
import asyncio

# Deepgram API Key from environment variable
DEEPGRAM_API_KEY = os.getenv('DEEPGRAM_API_KEY')

async def main():
    # Initialize Deepgram client
    dg_client = Deepgram(DEEPGRAM_API_KEY)

    # Options for real-time transcription
    options = {
        "keywords": ["JAY", "HAYWIRE"],
        "punctuate": True
    }

    # Connect to streaming endpoint
    async with dg_client.transcription.live(options) as stream:
        # Listen for the data
        async for data in stream:
            print("Transcript:", data['channel']['alternatives'][0]['transcript'])

# Run the main method
asyncio.run(main())

# Note: This script assumes you've set up a method to stream audio to Deepgram's API